docs/en/02_Developer_Guides/14_Files/01_File_Management.md

```diff
@@ -2,9 +2,122 @@ summary: Learn how to work with File and Image records
 
 # File Management
 
-## Files, Images and Folders as database records
+## Storage via database columns
 
-All files, images and folders in the 'assets' directory are stored in the database. Each record has the following database fields:
+Asset storage is provided out of the box via a [Flysystem](http://flysystem.thephpleague.com/) backed store.
+However, any class that implements the `AssetStore` interface could be substituted to provide storage backends
+via other mechanisms.
+
+Internally, files are stored as `[api:DBFile]` records on the rows of parent objects. These records are composite fields
+which contain sufficient information useful to the configured asset backend in order to store, manage, and 
+publish files. By default this composite field behind this field stores the following details:
+
+
+| Field name     | Description |
+| ----------     | -----------   
+| `Hash`         | The sha1 of the file content, useful for versioning (if supported by the backend) |
+| `Filename`     | The internal identifier for this file, which may contain a directory path (not including assets). Multiple versions of the same file will have the same filename. |
+| `Variant`      | The variant for this file. If a file has multiple derived versions (such as resized files or reformatted documents) then you can point to one of the variants here. |
+
+
+Note that the `Hash` and `Filename` always point to the original file, if a `Variant` is specified. It is up to the
```

Comment:
This should probably mention that files need to be managed exclusively via `AssetStore` - new files uploaded directly to e.g. S3 won't be picked up by it, and renamed or changed files risk losing their association with metadata stored through `DBFile`.